githooks/pre-commit

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+#
+# @file   pre-commit
+#         This hook script lints all code with Uncrustify, ESLint, and autopep8 on
```

Comment:
please document how to turn this feature off: bring all the condensed separate information below up here